### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by herve-perel**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [ ] Frontend
- [x] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [ ] I requested code review from other team members

## Frontend checklist:

- [ ] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [ ] My `Javascript` generate no new console errors
- [ ] I tested my code cross browsers
- [ ] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [ ] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [ ] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by Abecker37 — APPROVED** [review]
j'approuve la connexion et le register

**Review by bugBuster42 — APPROVED** [review]

**Review by Abecker37 — DISMISSED** [review]

**Review by Abecker37 — DISMISSED** [review]

**Review by bugBuster42 — APPROVED** [review]

**Review by JorisPineau — DISMISSED** [review]

File: packages/types/src/auth.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  lastname: string;
+  email: string;
+  password: string;
+  birthdate: Date;
+  biography?: string;
+  avatar?: string;
+  passenger_kilometer_traveled: number;
+  driver_kilometer_traveled: number;
+  economy_achieved: number;
+  created_at: Date;
```
Comment (YDrogen) [inline packages/types/src/auth.ts:12]: Pourquoi le `created_at` aurait besoin d'être dans le body ?

File: packages/types/src/auth.ts
Comment (YDrogen) [inline packages/types/src/auth.ts:1]: Il s'agit du Login ou du Register ? Si ce n'est pas les deux, il faut 2 types différents
ex: Pas besoin de rentrer la date de naissance pour la connexion

File: packages/types/src/auth.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  firstname: string;
+  lastname: string;
+  email: string;
+  password: string;
+  birthdate: Date;
+  biography?: string;
+  avatar?: string;
+  passenger_kilometer_traveled: number;
+  driver_kilometer_traveled: number;
+  economy_achieved: number;
```
Comment (YDrogen) [inline packages/types/src/auth.ts:11]: C'est l'utilisateur qui dit combien de km il veut au moment de s'inscrire ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by Abecker37 — APPROVED** [review]

**Pull request merged**